packages/mangrove.js/test/integration/offermaker.integration.test.ts

```diff
@@ -262,6 +276,37 @@ describe("OfferMaker", () => {
         );
       });
 
+      it("fails, when trying to cancel a non-existing offer", async () => {
+        const retractPromise = onchain_lp.retractBid(666, true); // with deprovision
+
+        assert.rejects(
+          retractPromise,
+          "Retracting a non-existing offer should fail."
+        );
+      });
+
+      it("fails, when trying to cancel an offer on a closed market", async () => {
+        const prov = await onchain_lp.computeBidProvision();
+
+        const { id: ofrId } = await onchain_lp.newBid({
+          wants: 10,
+          gives: 20,
+          fund: prov,
+        });
+
+        const base = onchain_lp.market.base.address;
+        const quote = onchain_lp.market.quote.address;
+        const closetx = await adminMgv.contract.deactivate(base, quote);
+        closetx.wait();
+
+        const retractPromise = onchain_lp.retractBid(ofrId, true); // with deprovision
+
+        assert.rejects(
+          retractPromise,
+          "Retracting on a closed market should fail."
+        );
+      });
```

Comment:
Canceling on closed markets is supposed to work, but we can adapt it to "creating on a closed market fails"

```suggestion
      it("fails, when trying to create an offer on a closed market", async () => {

        const base = onchain_lp.market.base.address;
        const quote = onchain_lp.market.quote.address;
        const closetx = await adminMgv.contract.deactivate(base, quote);
        await closetx.wait();

        const prov = await onchain_lp.computeBidProvision();

        const createPromise = onchain_lp.newAsk({
          wants: 10,
          gives: 20,
          fund: prov,
        });

        await assert.rejects(
          createPromise,
          "Creating on a closed semibook should fail."
        );
      });
```